pkg/net/net_state.go

```diff
@@ -0,0 +1,83 @@
+package net
+
+import (
+	"crypto/ecdsa"
+	"sync"
+
+	"nimona.io/pkg/crypto"
+	"nimona.io/pkg/net/peer"
+)
+
+type NetState interface {
+	AttachMandate(m *crypto.Mandate) error
+	AddAddress(addrs ...string)
+	GetPeerInfo() *peer.PeerInfo
+	GetPeerKey() *crypto.Key
+}
```

Comment:
`NetState` might not be the best name for this.
Addresses, mandate, and key are only needed to create the local peer info.
I know this isn't here to stay but cane we rename it to something like `LocalPeer`, `Node`, `Host` or something else?

Reply:
Actually, remove the interface, and use the `LocalInfo` everywhere instead maybe?

Reply:
done